rfcs/0007-multisigner-scripts.md

```diff
@@ -0,0 +1,128 @@
+- Feature Name: Multi-signer Scripts
+- Start Date: 2023-04-07
+- RFC PR: [FuelLabs/sway-rfcs#0000](https://github.com/FuelLabs/sway-rfcs/pull/001)
+- Sway Issue: [FueLabs/sway#0000](https://github.com/FuelLabs/sway/issues/001)
+
+# Summary
+
+[summary]: #summary
+
+This RFC proposes a new "Tx" struct/enum to be used in scripts to allow for atomic execution of bundle-like scripts.
+
+Allowing programmers to insert external transactions into their own scripts for atomic execution without any additional security assumptions can allow for a much more decentralised and secure MEV environment where both users and searchers benefit greatly, whilst also unlocking previously unseen types of MEV.
+
+# Motivation
+
+[motivation]: #motivation
+
+The current script features allow programmers to execute multiple transactions atomically, but it is limited to one party and cannot be extended to allow for atomic execution of transactions from multiple parties. Allowing insertion of third party transactions into scripts would remove the need for centralised block proposers who have privileged access to signed orderflow.
+
+# Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+The "Tx" enum comes in two variants, "Signed" and "Unsigned". The Signed variant represents a Transaction which has already been signed, and the Unsigned variant represents an intent of a transaction, without actually being signed yet.
+
+A script including a Signed Tx may execute the signed tx once at any point in their script, creating their own contract calls before or after the external tx. This can be used to arbitrage price changes, provide just in time liquidity, or other sorts of MEV tactics.
```

Comment:
```suggestion
A script including a Signed Tx may execute the signed tx once at any point in their script, creating their own contract calls before and/or after the external tx. This can be used to arbitrage price changes, provide just in time liquidity, or other sorts of MEV tactics.
```